libs/libcommon/src/libcommon/metrics.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import types
+from typing import Generic, Type, TypeVar
+
+from bson import ObjectId
+from mongoengine import Document
+from mongoengine.fields import DateTimeField, DictField, ObjectIdField, StringField
+from mongoengine.queryset.queryset import QuerySet
+
+from libcommon.constants import METRICS_MONGOENGINE_ALIAS, METRICS_TTL_SECONDS
+from libcommon.utils import get_datetime
+
+# START monkey patching ### hack ###
+# see https://github.com/sbdchd/mongo-types#install
+U = TypeVar("U", bound=Document)
+
+
+def no_op(self, x):  # type: ignore
+    return self
+
+
+QuerySet.__class_getitem__ = types.MethodType(no_op, QuerySet)
+
+
+class QuerySetManager(Generic[U]):
+    def __get__(self, instance: object, cls: Type[U]) -> QuerySet[U]:
+        return QuerySet(cls, cls._get_collection())
+
+
+# END monkey patching ### hack ###
+
+
+class CustomMetric(Document):
+    """A response computed for a job, cached in the mongoDB database
+
+    Args:
+        metric (`str`): metric name
+        content (`dict`): The content of the metric.
+        created_at (`datetime`): When the metric has been created.
+    """
+
+    id = ObjectIdField(db_field="_id", primary_key=True, default=ObjectId)
+
+    metric = StringField(required=True)
+    content = DictField(required=True)
+    created_at = DateTimeField(default=get_datetime)
+
+    meta = {
+        "collection": "customMetrics",
+        "db_alias": METRICS_MONGOENGINE_ALIAS,
+        "indexes": [
+            ("metric", "created_at"),
+            {"fields": ["created_at"], "expireAfterSeconds": METRICS_TTL_SECONDS},
```

Comment:
I'm not sure we want to delete the metrics automatically. Why is that?

Reply:
OK. It's because you create new entries on every creation of metrics, and the old ones are deleted. And this is only to clean the house in case some old entries have not been deleted. OK